parcels/particleset/particlesetsoa.py

```diff
@@ -427,6 +489,36 @@ def to_dict(self, pfile, time, deleted_only=False):
         return self._collection.toDictionary(pfile=pfile, time=time,
                                              deleted_only=deleted_only)
 
+    def compute_neighbor_tree(self, time, dt):
+        active_mask = self.active_particles_mask(time, dt)
+
+        self._values = np.vstack((
+            self._collection.data['depth'],
+            self._collection.data['lat'],
+            self._collection.data['lon'],
+        ))
```

Comment:
Does that allocate a new array and deep-copy those fields ? If that is so, that seems to be a huge memory sink here. Just saying.

Reply:
Yes.. future possible improvement I suppose?

Reply:
well, I guess I should make a note of that somewhere then ...